homeassistant/components/slow_pwm/number.py

```diff
@@ -0,0 +1,410 @@
+"""Number entity containing a slow_pwm number."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from math import isnan
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.number import (
+    DEFAULT_MAX_VALUE,
+    DEFAULT_MIN_VALUE,
+    DEFAULT_STEP,
+    PLATFORM_SCHEMA,
+    RestoreNumber,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_MAXIMUM,
+    CONF_MINIMUM,
+    CONF_MODE,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+    EVENT_HOMEASSISTANT_START,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.core import CoreState, HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import (
+    async_track_point_in_utc_time,
+    async_track_time_interval,
+)
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+import homeassistant.util.dt as dt_util
+
+from . import PLATFORMS
+from .const import (
+    ATTR_CYCLE_TIME,
+    ATTR_LAST_CYCLE_START,
+    ATTR_MIN_SWITCH_TIME,
+    ATTR_OUTPUT_STATES,
+    ATTR_TIMED_OUTPUT,
+    CONF_CYCLE_TIME,
+    CONF_MIN_SWITCH_TIME,
+    CONF_OUTPUTS,
+    CONF_STEP,
+    DEFAULT_CYCLE_TIME,
+    DEFAULT_MODE,
+    DEFAULT_SWITCH_TIME,
+    DOMAIN,
+    MODE_AUTO,
+    MODE_BOX,
+    MODE_SLIDER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>